tools/tasks/seed/check.tools.ts

```diff
@@ -0,0 +1,30 @@
+import * as util from 'gulp-util';
+import * as runSequence from 'run-sequence';
+
+import { join } from 'path';
+
+let fs = require('fs');
+
+
+// if gulpfile.ts has been compiled then we need to rebuild the toolchain
+
+export = (done: any) => {
+
+  let checkFile = join(process.cwd(), 'tools', 'config.js');
+
+  // need to require the build.toolchain task as it won't be able to run after we run clear.files
+  let buildTools = require('./build.tools');
+
+  try {
+    fs.accessSync(checkFile, fs.F_OK);
+    util.log('Gulpfile has previously been compiled, rebuilding toolchain');
+    runSequence('clean.tools');
```

Comment:
ah i see what you mean now. 'build.tools' can't be in the run sequence as the 'clean.tools' deletes the compiled files for build.tools, so when it attempts to run it fails with the build.tools.js doesn't exist error. if clean.tools was async then we could wait for the promise to resolve before continuing i guess. 

i note your previous commit breaks del, and possibly also rimraf, i'm getting unable to compile typescript errors:

tools/tasks/seed/clear.files.ts (23,22): Argument of type 'string[]' is not assignable to parameter of type 'string'. (2345)

for rimraf and the typings for del doesn't exist anymore